lib/elixir/lib/enum.ex

```diff
@@ -2484,8 +2484,8 @@ defmodule Enum do
 
   ## shuffle
 
-  defp unwrap([{_, h} | collection], t) do
-    unwrap(collection, [h|t])
```

Comment:
agreed.
yes, this was part of the big commit yesterday, i just split it to make it easy to review.

I will leave a comment in a few lines bellow, you make consider it split them due to their length